database/rrdset.c

```diff
@@ -424,6 +424,9 @@ void rrdset_delete(RRDSET *st) {
     }
 
     recursively_delete_dir(st->cache_dir, "left-over chart");
+#ifdef ENABLE_ACLK
+    aclk_del_collector(st->rrdhost->hostname, st->plugin_name, st->module_name);
```

Comment:
Just a general comment. I would maybe add a generic `del_collector_notif` function def and then here call all of those functions that registered.
The reasoning behind this is that RRDHOT/RRDSET/RRDDIM are already interdependent with all code making it quite a spaghetti.
That way RRDSET doesn't have to know about existence of some `ACLK` but will just call all registered callbacks of same function type.

Reply:
It would be nice to have callbacks indeed. Luckily in this case we only try to delete a collector when a chart is deleted and the (host, plugin_name, module_name) combination ends up having 0 charts registered

Reply:
yes this comment applies to multiple similar places where now RRDXYZ is dependent on ACLK... but maybe idea to improve in future. I will not consider this to be a blocker for merge of this PR

Reply:
Please don't do this - while it would make the code more generic we have no use-cases that warrant it. If a use-case is added in the future and there multiple notification to send out then it would make sense to add the callback that you are describing - but currently it adds complexity with no benefit.